src/compiler/program.ts

```diff
@@ -1059,6 +1064,10 @@ namespace ts {
                 else {
                     // Compute the commonSourceDirectory from the input files
                     commonSourceDirectory = computeCommonSourceDirectory(files);
+                    // If we failed to find a good common directory, but outDir is specified and at least one of our files is on a windows drive/URL/other resource, add a failure
+                    if (options.outDir && commonSourceDirectory === "" && forEach(files, file => getRootLength(file.fileName) > 1)) {

```

Comment:
Correct - in effect, we want to allow `""` as a valid common source directory provided either all paths are relative or all paths are rooted to a unix-style drive (with a root of `/`), otherwise `""` is the return value when there is no available common path and we should error in its presence.